external/storm-kafka-client/src/main/java/org/apache/storm/kafka/spout/KafkaSpout.java

```diff
@@ -289,6 +293,13 @@ private void doSeekRetriableTopicPartitions() {
                 kafkaConsumer.seek(rtp, acked.get(rtp).getCommittedOffset() + 1);    // Seek to last committed offset
             }
         }
+        
+        if(!retriableTopicPartitions.isEmpty()) {
+            //Pause other partitions for this poll so maxUncommittedPartitions isn't exceeded by too much
```

Comment:
I am a bit confused about this. When can it be exceeded? By how much? This should be a deterministic number.

I read your comment in the email thread about  maxSpoutPending. Is this related to that ?